src/NHibernate/Linq/Visitors/WhereJoinDetector.cs

```diff
@@ -119,174 +119,164 @@ public override Expression VisitExpression(Expression expression)
-				_values.Pop();
-
-				HandleUnaryOperation(pvs => pvs.IsNull());
-			}
-			else if (expression.NodeType == ExpressionType.Equal && VisitorUtil.IsNullConstant(expression.Left))
-			{
-				// Discard result from left null.
-				var right = _values.Pop();
-				_values.Pop(); // Discard left.
-				_values.Push(right);
-
-				HandleUnaryOperation(pvs => pvs.IsNull());
-			}
-			else if (expression.NodeType == ExpressionType.Coalesce)
-			{
-				HandleBinaryOperation((a, b) => a.Coalesce(b));
-			}
-			else if (expression.NodeType == ExpressionType.Add || expression.NodeType == ExpressionType.AddChecked)
-			{
-				HandleBinaryOperation((a, b) => a.Add(b, expression.Type));
-			}
-			else if (expression.NodeType == ExpressionType.Divide)
-			{
-				HandleBinaryOperation((a, b) => a.Divide(b, expression.Type));
-			}
-			else if (expression.NodeType == ExpressionType.Modulo)
-			{
-				HandleBinaryOperation((a, b) => a.Modulo(b, expression.Type));
-			}
-			else if (expression.NodeType == ExpressionType.Multiply || expression.NodeType == ExpressionType.MultiplyChecked)
-			{
-				HandleBinaryOperation((a, b) => a.Multiply(b, expression.Type));
-			}
-			else if (expression.NodeType == ExpressionType.Power)
-			{
-				HandleBinaryOperation((a, b) => a.Power(b, expression.Type));
-			}
-			else if (expression.NodeType == ExpressionType.Subtract || expression.NodeType == ExpressionType.SubtractChecked)
-			{
-				HandleBinaryOperation((a, b) => a.Subtract(b, expression.Type));
-			}
-			else if (expression.NodeType == ExpressionType.And)
-			{
-				HandleBinaryOperation((a, b) => a.And(b, expression.Type));
-			}
-			else if (expression.NodeType == ExpressionType.Or)
-			{
-				HandleBinaryOperation((a, b) => a.Or(b, expression.Type));
-			}
-			else if (expression.NodeType == ExpressionType.ExclusiveOr)
-			{
-				HandleBinaryOperation((a, b) => a.ExclusiveOr(b, expression.Type));
-			}
-			else if (expression.NodeType == ExpressionType.LeftShift)
-			{
-				HandleBinaryOperation((a, b) => a.LeftShift(b, expression.Type));
-			}
-			else if (expression.NodeType == ExpressionType.RightShift)
-			{
-				HandleBinaryOperation((a, b) => a.RightShift(b, expression.Type));
-			}
-			else if (expression.NodeType == ExpressionType.Equal)
-			{
-				HandleBinaryOperation((a, b) => a.Equal(b));
-			}
-			else if (expression.NodeType == ExpressionType.NotEqual)
-			{
-				HandleBinaryOperation((a, b) => a.NotEqual(b));
-			}
-			else if (expression.NodeType == ExpressionType.GreaterThanOrEqual)
-			{
-				HandleBinaryOperation((a, b) => a.GreaterThanOrEqual(b));
-			}
-			else if (expression.NodeType == ExpressionType.GreaterThan)
-			{
-				HandleBinaryOperation((a, b) => a.GreaterThan(b));
-			}
-			else if (expression.NodeType == ExpressionType.LessThan)
-			{
-				HandleBinaryOperation((a, b) => a.LessThan(b));
-			}
-			else if (expression.NodeType == ExpressionType.LessThanOrEqual)
-			{
-				HandleBinaryOperation((a, b) => a.LessThanOrEqual(b));
```

Comment:
Huge pile of `else if` instead of a switch case. Not really related to the migration, but well, with the amount of other changes on those classes, I have decide to refactor such things by the way.